precompile/contracts/rewardmanager/contract.go

```diff
@@ -45,34 +45,31 @@ var (
 	//go:embed contract.abi
 	RewardManagerRawABI string
 
-	RewardManagerABI        abi.ABI                                // will be initialized by init function
-	RewardManagerPrecompile precompile.StatefulPrecompiledContract // will be initialized by init function
+	RewardManagerABI        = parseABI(RewardManagerRawABI)
+	RewardManagerPrecompile = createRewardManagerPrecompile() // will be initialized by init function
 
 	rewardAddressStorageKey        = common.Hash{'r', 'a', 's', 'k'}
 	allowFeeRecipientsAddressValue = common.Hash{'a', 'f', 'r', 'a', 'v'}
 )
 
-func init() {
-	parsed, err := abi.JSON(strings.NewReader(RewardManagerRawABI))
-	if err != nil {
-		panic(err)
-	}
-	RewardManagerABI = parsed
-	RewardManagerPrecompile, err = createRewardManagerPrecompile(precompile.RewardManagerAddress)
+func parseABI(rawABI string) abi.ABI {
```

Comment:
Can we put this into a utility, so that it's not duplicated across precompile contracts?

Reply:
yea good call, moved